src/app/common/hooks/auth/use-magic-recovery-code.ts

```diff
@@ -1,90 +1,79 @@
-import { FormEvent, useCallback, useEffect, useState } from 'react';
-import { useNavigate } from 'react-router-dom';
-import { decrypt } from '@stacks/wallet-sdk';
+import { useCallback, useEffect, useState } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
 
 import { useLoading } from '@app/common/hooks/use-loading';
 import { useWallet } from '@app/common/hooks/use-wallet';
 import { useOnboardingState } from '@app/common/hooks/auth/use-onboarding-state';
 import { RouteUrls } from '@shared/route-urls';
-import {
-  useMagicRecoveryCodePasswordState,
-  useMagicRecoveryCodeState,
-} from '@app/store/onboarding/onboarding.hooks';
+import { decrypt } from '@stacks/wallet-sdk';
+import { useDispatch } from 'react-redux';
+import { keyActions } from '@app/store/keys/key.actions';
+import { delay } from '@app/common/utils';
+import toast from 'react-hot-toast';
+
+function pullMagicRecoveryCodeFromParams(urlSearchParams: URLSearchParams) {
+  return urlSearchParams.get('magicRecoveryCode');
+}
+
+async function simulateShortDelayToAvoidImmediateNavigation() {
```

Comment:
I love this name ❤️

Reply:
Kyran taught me that very descriptive names lead to instant understanding

Reply:
❤️ 

Better this than 

```ts
// added this delay to avoid immediate navigation
await delay(500);
```